File: apps/desktop/src/stores/preferences/live.ts
/**
 * Live sessions preferences slice (ADR-0016).
 *
 * Owns whether TracePilot attaches automatically to sessions running in a
 * `copilot --ui-server` terminal, and whether the terminals it launches or
 * resumes get `--ui-server` so they can be attached.
 */

import type { TracePilotConfig } from "@tracepilot/types";
import { ref } from "vue";

type LiveConfig = TracePilotConfig["live"];

export function createLiveSlice() {
  /** Attach automatically to attachable sessions when they are opened. */
  const liveAutoAttach = ref(true);
  /** Launch and resume terminals with `--ui-server`. */
  const liveLaunchAttachable = ref(true);

  function hydrateLive(config: Partial<LiveConfig> | undefined) {
    liveAutoAttach.value = config?.autoAttach ?? true;
    liveLaunchAttachable.value = config?.launchAttachable ?? true;
  }

  function buildLiveConfig(): LiveConfig {
    return {
      autoAttach: liveAutoAttach.value,
      launchAttachable: liveLaunchAttachable.value,
    };
  }

  return { liveAutoAttach, liveLaunchAttachable, hydrateLive, buildLiveConfig };
}

File: crates/tracepilot-tauri-bindings/src/config/live.rs
//! Live session preferences (Copilot SDK live attach, ADR-0016).

use serde::{Deserialize, Serialize};

use super::defaults::default_true;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LiveConfig {
    /// Attach automatically when a session view opens on a session that is
    /// running in an attachable (`--ui-server`) terminal. Each attach appends
    /// one `session.resume` event to the session's history.
    #[serde(default = "default_true")]
    pub auto_attach: bool,
    /// Start terminals that TracePilot launches or resumes with `--ui-server`
    /// so they can be attached to. Only applies while live sessions
    /// (`features.copilotSdk`) are enabled.
    #[serde(default = "default_true")]
    pub launch_attachable: bool,
}

impl Default for LiveConfig {
    fn default() -> Self {
        Self {
            auto_attach: true,
            launch_attachable: true,
        }
    }
}
